ios/Video/Features/RCTPlayerObserver.swift

```diff
@@ -50,9 +50,14 @@ class RCTPlayerObserver: NSObject {
             removePlayerItemObservers()
         }
         didSet {
-            if playerItem != nil {
-                addPlayerItemObservers()
-            }
+            guard let playerItem = playerItem else { return }
+
+            addPlayerItemObservers()
+
+            // handle timedMetadata
+            let metadataOutput = AVPlayerItemMetadataOutput()
+            playerItem.add(metadataOutput)
+            metadataOutput.setDelegate(self, queue: .global())
```

Comment:
Are you 100% of the .global(). In my understanding, it will handle the update in UI thread, if possible I think we should move it to another thread. Thanks !

Reply:
![CleanShot 2023-12-28 at 13 03 28@2x](https://github.com/react-native-video/react-native-video/assets/83216158/e89fd882-d587-43f7-878f-4de854425f73)
If i undterstand this correctly, `.global()` is what we want to use. `.main` will be for UI

Reply:
Every example I found was actually using the main UI queue. https://dcordero.medium.com/hls-timed-metadata-with-avplayer-9e20806ef92f

If you intend to use background queue specify quality of service so those calls get higher priority

Reply:
Done, moved to main queue ✅